docs/proposals/20230407-managed-k8s-capi-contract-changes.md

```diff
@@ -0,0 +1,380 @@
+- `<Infra>ControlPlane`
+  - When we say `<Infra>ControlPlane` we refer to any provider's infra-specific implementation of the a Kubernetes cluster's control plane. When you see `<Infra>`, interpret that as a placeholder for any provider implementation. Some concrete examples of provider infra control plane implementations are Azure's CAPZ provider (e.g., `AzureManagedControlPlane`), AWS's CAPA provider (e.g., `AWSManagedControlPlane`), and Google Cloud's CAPG provider (e.g., `GCPManagedControlPlane`).
+- Managed Kubernetes
+  - Managed Kubernetes refers to any Kubernetes Cluster provisioning and maintenance platform that is exposed by a service API. For example: [EKS](https://aws.amazon.com/eks/), [OKE](https://www.oracle.com/cloud/cloud-native/container-engine-kubernetes/), [AKS](https://azure.microsoft.com/en-us/products/kubernetes-service), [GKE](https://cloud.google.com/kubernetes-engine), [IBM Cloud Kubernetes Service](https://www.ibm.com/cloud/kubernetes-service), [DOKS](https://www.digitalocean.com/products/kubernetes), and many more throughout the Kubernetes Cloud Native ecosystem.
+- _Kubernetes Cluster Infrastructure_
+  - When we refer to _Kubernetes Cluster Infrastructure_ we aim to distinguish required environmental infrastructure (e.g., cloud virtual networks) in which a Kubernetes cluster resides as a "set of child resources" from the Kubernetes cluster resources themselves (e.g., virtual machines that underlie nodes, managed by Cluster API). Sometimes this is referred to as "BYO Infrastructure"; essentially, we are talking about **infrastructure that supports a Kubernetes cluster, but is not actively managed by Cluster API**. As we will see, this boundary is different when discussing Managed Kubernetes: more infrastructure resources are not managed by Cluster API when running Managed Kubernetes.
+- e.g.
+  - This just means "For example:"!
+
+## Summary
+
+We propose to make provider `<Infra>Cluster` resources optional in order to better represent Managed Kubernetes scenarios where all _Kubernetes Cluster Infrastructure_ is managed by the service provider, and not by Cluster API. In order to support that, we propose that the API Server endpoint reference can also originate from the `<Infra>ControlPlane` resource, and not the `<Infra>Cluster` resource. These changes will introduce two new possible implementation options for providers implementing Managed Kubernetes in Cluster API:
```

Comment:
Is there any recommendation or consideration we want to capture about on how this might impact machine management and infraMachine controller vs a infraMachinePool implementation choices?

Reply:
just wondering if you had drawn some conclusions about existing patterns and future recommendations while fleshing out this proposal, e.g in AWS the pattern scope.IsEKSManaged() is used and ignores the infraCluster resource, this is something we might / might not perpetuate.
sgtm if we don't get into this discussion here. Thanks.